### PR TITLE
feat: add starting index and event to fire when transition is finished

### PR DESCRIPTION
**Pull request by tfriedlander**
### Short Description

Adds the ability to specify starting index and a callback on transition completion.

### Use case

I use this component to make a simple page with little to no content on it shine. It is fantastic and easy to use. One problem I am encountering is that I want to be able to know what image is being shown by other components. 

When using it in conjunction with other components we may want to have communication between the slideshow and a state management (such as [redux](https://redux.js.org/)). I think this would be useful when wanting to maintain the currently shown background image in state. Exposing a callback allows us to use an action to update this information. This allows other components to act differently based upon which image is shown. 

### Example

```js
import React, { Component } from 'react'

import ReactBackgroundSlideshow from 'react-background-slideshow'

import image1 from './assets/001.jpg'
import image2 from './assets/002.jpg'
import image3 from './assets/003.jpg'
import image4 from './assets/004.jpg'
import image5 from './assets/005.jpg'
import image6 from './assets/006.jpg'

const images = [
  image1,
  image2,
  image3,
  image4,
  image5,
  image6
]

const params = {
  images,
  onChange: (attrs) => console.log('image changed', attrs),
  startAt: 3,
}

export default class App extends Component {
  render () {
    return (
      <ReactBackgroundSlideshow {...params} />
    )
  }
}
```
image

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -36,7 +38,7 @@ export default class ReactBackgroundSlideshow extends Component {
   state = {
     isAnimating: false,
     direction: 'next',
-    current: Math.random() * this.props.images.length | 0,
+    current: this.props.startAt || (Math.random() * this.props.images.length | 0),
```
Comment (transitive-bullshit) [inline src/index.js]: This won't do what you want if `startAt` is 0. :)

**Comment by transitive-bullshit** [comment]
@tfriedlander this change looks great! Just one minor issue and I'll merge :)

**Comment by tfriedlander** [comment]
Ah, I missed that. Thanks, nice catch!

**Comment by transitive-bullshit** [comment]
Let me know once you've made that change. Also, please star this repo if it's helped you out to help spread the word! 👍

**Comment by tfriedlander** [comment]
~Should be good now (hopefully 😸). Thanks very much!~ oops, removing accidental test case in the example then should be good

**Comment by transitive-bullshit** [comment]
Published in `v1.0.4`

**Pull request merged**